app/models/custom_hint.rb

```diff
@@ -64,7 +65,13 @@ def validate_csv
   def process_records
     validate_csv
     CSV.parse(@csv, headers: true) do |record|
-      if %w(Title URL Fingerprint).none? { |header| record[header].nil? }
+      if %w(Title URL).all? { |header| record[header].present? }
+        # Add the fingerprint if needed; people creating spreadsheet rows may
```

Comment:
I think you're right - I've been keeping them as a historical artifact (because they happened to be part of the original spreadsheet), but there's not logical value in keeping them going forward, and the code can certainly be simplified without them.